subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/impl/DefaultFileSystemSnapshotterTest.groovy

```diff
@@ -243,6 +257,217 @@ class DefaultFileSystemSnapshotterTest extends Specification {
+
+        then:
+        assertSingleFileSnapshot(snapshots)
+
+
+        when:
+        snapshots = snapshotter.snapshot(fromConfigurableFiles.matching { true })
+
+        then:
+        assertSingleFileSnapshot(snapshots)
+
+
+        when:
+        snapshots = snapshotter.snapshot(fromConfigurableFiles.matching { false })
+
+        then:
+        assertSingleFileSnapshot(snapshots)
+    }
+
+    def "snapshots a archive trees as RegularFileSnapshot"() {
+        given:
+        def tempDir = tmpDir.createDir('tmpDir')
+        def archiveBaseDir = tempDir.createDir('archiveBase')
+        def file = archiveBaseDir.createFile('file.txt')
+        file.text = "content"
+
+        when:
+        TestFile zip = tempDir.file('archive.zip');
+        archiveBaseDir.zipTo(zip)
+        def zipTree = TestFiles.fileOperations(tempDir, testFileProvider()).zipTree(zip)
+        def snapshots = snapshotter.snapshot(zipTree)
+
+        then:
+        assertSingleFileSnapshot(snapshots)
+
+        when:
+        TestFile tar = tempDir.file('archive.tar');
+        archiveBaseDir.tarTo(tar)
+        def tarTree = TestFiles.fileOperations(tempDir, testFileProvider()).tarTree(tar)
+        snapshots = snapshotter.snapshot(tarTree)
+
+        then:
+        assertSingleFileSnapshot(snapshots)
+
+        when:
+        def tarDir = tmpDir.createDir('tarDir')
+        TestFile emptyTar = tempDir.file('emptyArchive.tar');
+        tarDir.tarTo(emptyTar)
+        def emtpyTarTree = TestFiles.fileOperations(tempDir, testFileProvider()).tarTree(tar)
+        snapshots = snapshotter.snapshot(emtpyTarTree)
+
+        then:
+        assertSingleFileSnapshot(snapshots)
+
+        when:
+        def tgzDir = tmpDir.createDir('tgzDir')
+        TestFile tgz = tempDir.file('emptyArchive.tgz');
+        tgzDir.tgzTo(tgz)
+        def localResource = new LocalResourceAdapter(TestFiles.fileRepository().localResource(tgz));
+        def emtpyTgzTree = TestFiles.fileOperations(tempDir, testFileProvider()).tarTree(localResource)
+        snapshots = snapshotter.snapshot(emtpyTgzTree)
+
+        then:
+        assertSingleFileSnapshot(snapshots)
+
+        when:
+        def readableResource = new ReadableResource() {
+            @Override
+            InputStream read() throws MissingResourceException, ResourceException {
+                return tgz.newInputStream()
+            }
+
+            @Override
+            String getDisplayName() {
+                return tgz.getName()
+            }
+
+            @Override
+            URI getURI() {
+                return tgz.toURI()
+            }
+
+            @Override
+            String getBaseName() {
+                return tgz.getName()
+            }
+        };
+        def recourceTarTree = TestFiles.fileOperations(tempDir, testFileProvider()).tarTree(readableResource)
+        snapshots = snapshotter.snapshot(recourceTarTree)
+
+        then:
+        assertSingleFileSnapshot(snapshots)
```

Comment:
So the assertion fails as I expected: https://builds.gradle.org/viewLog.html?buildId=16913443&buildTypeId=Gradle_Check_Stage_QuickFeedbackLinuxOnly_Trigger

And that is good! So let's fix up the assertion.

Reply:
fixed